clustered/client/src/main/java/org/ehcache/clustered/config/ClusteringServiceConfiguration.java

```diff
@@ -32,21 +37,45 @@
  */
 // TODO: Should this accept/hold a *list* of URIs?
 // TODO: Add validation for connection URI(s)
-public class ClusteringServiceConfiguration
+public final class ClusteringServiceConfiguration
     implements ServiceCreationConfiguration<ClusteringService>,
     CacheManagerConfiguration<PersistentCacheManager> {
 
-  private final URI connectionUrl;
+  private final URI clusterUri;
+  private final String defaultPool;
+  private final Map<String, PoolDefinition> pools;
+
+  public ClusteringServiceConfiguration(final URI clusterUri, String defaultPool, Map<String, PoolDefinition> pools) {
+    if (clusterUri == null) {
+      throw new NullPointerException("connectionUrl");
+    }
+    if (defaultPool == null) {
+      throw new NullPointerException("defaultPool");
+    }
+    this.clusterUri = clusterUri;
+    this.defaultPool = defaultPool;
+    this.pools = unmodifiableMap(new HashMap<String, PoolDefinition>(pools));
+  }
 
-  public ClusteringServiceConfiguration(final URI connectionUrl) {
-    if (connectionUrl == null) {
+  public ClusteringServiceConfiguration(URI clusterUri, Map<String, PoolDefinition> pools) {
+    if (clusterUri == null) {
       throw new NullPointerException("connectionUrl");
     }
-    this.connectionUrl = connectionUrl;
+    this.clusterUri = clusterUri;
+    this.defaultPool = null;
+    this.pools = unmodifiableMap(new HashMap<String, PoolDefinition>(pools));
   }
 
   public URI getConnectionUrl() {
```

Comment:
Should `getConnectionUrl` be renamed to match the field and constructor parameter?